docs/pages/identity-governance/entra-id/terraform.mdx

```diff
@@ -0,0 +1,378 @@
+---
+title:  Configure Entra ID integration using Terraform
+description: Describes how to set up Teleport Entra ID integration using Terraform
+sidebar_position: 3
+labels:
+ - how-to
+ - identity-governance
+ - terraform
+---
+
+This guide shows Entra ID integration configuration using Terraform and `tctl`.
+
+See [getting started with Entra ID integration](getting-started.mdx) for a guided Entra ID configuration. 
+
+## Prerequisites
+
+- Teleport Identity Governance enabled for your Teleport cluster.
+- Your user must have privileged administrator permissions in Microsoft Azure and Entra ID tenant.
+- Terraform configured with the `azuread` [provider](https://registry.terraform.io/providers/hashicorp/azuread/latest/docs).
+
+## Step 1/4. Create enterprise application
+
+This application will be used to configure Entra ID based SSO in Teleport.
+```hcl
+resource "azuread_application" "example" {
+  display_name = "example"
+
+  # These resources will be updated later and should not 
+  # be overridden.
+  lifecycle {
+    ignore_changes = [
+      identifier_uris,
+      required_resource_access,
+    ]
+  }
+
+  feature_tags {
+    enterprise            = true
+    custom_single_sign_on = true
+  }
+}
+```
+
+
+## Step 2/4. Configure SAML SSO
+
+Configure Entra ID enterprise application as an SAML SSO provider for Teleport.
+```hcl
+locals {
+  # Enter your Teleport cluster address. 
+  # E.g. format of the expected value: https://example.teleport.sh/v1/webapi/saml/acs/entra-id
+  teleport_saml_entity_id = "https://<Var name="example.teleport.sh"/>/v1/webapi/saml/acs/entra-id"
+}
+
+resource "azuread_service_principal" "example_sp" {
+  client_id = azuread_application.example.client_id
+
+  preferred_single_sign_on_mode = "saml"
+
+  login_url = local.teleport_saml_entity_id
+
+  feature_tags {
+    enterprise            = true
+    custom_single_sign_on = true
+  }
+}
+```
+
+Configure a certificate to be used for SAML assertion signing.
+```hcl
+resource "azuread_service_principal_token_signing_certificate" "example_saml_cert" {
+  service_principal_id = azuread_service_principal.example_sp.id
+  display_name         = "CN=teleport-tf-example-sso"
```

Comment:
What is the `teleport-tf-example-sso` here? Is this something users would need to override?